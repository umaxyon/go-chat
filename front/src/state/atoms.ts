import { atom } from "recoil"

export type Member = {
    user: string
}

export type FeedRow = {
    id: string,
    MessageType: "comment" | "addMember" | "leaveMember" | "system",
    user: string,
    text: string,
    createdAt: any
}

export type LoginData = {
    user: string,
    token: string,
    disconnect?: () => void
}

export const membersState = atom<Member[]>({
    key: "members",
    default: [],
})

export const commentsState = atom<FeedRow[]>({
    key: "comments",
    default: [],
})

export const loginState = atom<LoginData>({
    key: "loginUser",
    default: undefined,
})