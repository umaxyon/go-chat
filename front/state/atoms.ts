import { atom } from "recoil"

export type Member = {
    user: string
}

export type Comment = {
    id: string,
    user: string,
    text: string,
    createdAt: any
}

export type LoginData = {
    user: string,
    disconnect?: () => void
}

export const membersState = atom<Member[]>({
    key: "members",
    default: [],
})

export const commentsState = atom<Comment[]>({
    key: "comments",
    default: [],
})

export const loginState = atom<LoginData>({
    key: "loginUser",
    default: undefined,
})