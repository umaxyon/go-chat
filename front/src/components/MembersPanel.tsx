import { useRecoilValue } from "recoil"
import { membersState } from "../state/atoms"


const MembersPanel = () => {
    const members = useRecoilValue(membersState)

    return (
        <div className="w-full h-full p-1 overflow-y-scroll overflow-x-hidden scrollDiv">
            {members.map((m, i) => { return (
                <div key={`${m.user}_${i}`} className="text-sm">{m.user}</div>
            )})}
        </div>
    )
}
export default MembersPanel
