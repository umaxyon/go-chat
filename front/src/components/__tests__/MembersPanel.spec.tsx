import MembersPanel from "@/components/MembersPanel"
import "@testing-library/jest-dom/extend-expect";
import { cleanup } from '@testing-library/react';
import { useRecoilValue } from 'recoil'
import renderer from "react-test-renderer"
import { Member } from "src/state/atoms";

afterEach(() => cleanup())

jest.mock("recoil");

const setRecoilMock = (val: Member[]) => {
    (useRecoilValue as jest.Mock).mockImplementation(() => val)
}

const sut = (<MembersPanel />)

describe('MembersPanelテスト', () => {
    it('snapshot test', () => {
        const testMembers = [
            { user: "aaa" }, { user: "bbb" }, { user: "ccc" }
        ]
        setRecoilMock(testMembers)

        const component = renderer.create(sut);
        let tree = component.toJSON()
        expect(tree).toMatchSnapshot()
    })
})




