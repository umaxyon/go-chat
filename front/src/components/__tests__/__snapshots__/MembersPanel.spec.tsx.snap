// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MembersPanelテスト snapshot test 1`] = `
<div
  className="w-full h-full p-1"
>
  <div
    className="text-sm"
  >
    aaa
  </div>
  <div
    className="text-sm"
  >
    bbb
  </div>
  <div
    className="text-sm"
  >
    ccc
  </div>
</div>
`;
